tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/wizard/template/CDIWebProjectWizardTemplate.java

```diff
@@ -101,6 +85,8 @@ public void createCDIProject(){
 	//cd1.0 should show warning
 	@Test
 	public void createCDIProjectWithoutBeansXml(){
+		cleanUpTheWorkspace();
```

Comment:
If you want to make sure that there is no project in the begging of the test (test class) you may use CleanWorkplace requirement instead of calling after method which does the same in other way.

Reply:
I agree, fixed.

Reply:
And yet, you still are using cleanUpTheWorkspace in before which is basically the same. You are cleaning up the workspace multiple times in between test executions. In before and after methods.

What I think you have defined here, workflow of what is being called when running UtilityProjectWithCDITestCDI20 test class:

- CDITestBase.Before
- ProjectWithCDITemplate.Before
- UtilityProjectWithCDITestCDI20.Before -> Here you call what it also defined in CDITestBase.AfterClass - Cleanup
- UtilityProjectWithCDITestCDI20.Test1
- UtilityProjectWithCDITestCDI20.After
- ProjectWithCDITemplate.After  (CleanupWorkspace)
- CDITestBase.After
- CDITestBase.Before
- ProjectWithCDITemplate.Before
- UtilityProjectWithCDITestCDI20.Before -> Here you call what it also defined in CDITestBase.AfterClass - Cleanup
- UtilityProjectWithCDITestCDI20.Test2
- UtilityProjectWithCDITestCDI20.After
- ProjectWithCDITemplate.After  (CleanupWorkspace)
- CDITestBase.After

CleanUpTheWorkspace = CleanUp from base class. 
What I suggest it to keep only one definition of the method which does clean up. Do not call it in before class of "the most childish" class as it is being called in After method of the middle class (Template). And to make sure that workspace is cleaned up before running test class, use requirement CleanWorkspace from reddeer.

Reply:
You are right, the cleaning process runs several times. It does not make sense to keep it the way it is now. Fixed.